### ISSUE TITLE
"Failed to fetch dynamically imported module" when starting the game

### ISSUE THREAD

**Issue by arnehaber** [danielyxie/bitburner#2706]
After starting the game, I get the following error for all of my running scripts:

```
RUNTIME ERROR
hackOnly.js@CSEC
Args: ["neo-net", 1]

Failed to fetch dynamically imported module: blob:file:///5dfb196d-aca1-4ead-b204-944ecdb4f0b5
stack:
TypeError: Failed to fetch dynamically imported module: blob:file:///5dfb196d-aca1-4ead-b204-944ecdb4f0b5
```

My script that throws the error is pretty simple:
```
import { logFromTo } from "helpers.js"; 

/** @param {NS} ns **/
export async function main(ns) {
	var server = ns.args[0];
	var hackThreads = ns.args[1];	
	logFromTo(ns, ns.getHackTime(server));
	await ns.hack(server, { threads: hackThreads })
}
```

The helper functions file "helpers.js" is located next to the running script.
If it matters, here is the content:

```
/** @param {NS} ns **/
export async function main(ns) {
	ns.tprint(printTimeStamp());
}

export function notice(ns, msg) {
	ns.print(`[${printTimeStamp()}] ${msg}`);
	ns.print(msg);
}

export function printTimeStamp() {
	return dateToString(Date.now());
}

export function logFromTo(ns, durationInMs) {
	var startTime = Date.now();
	var expectedResultAt = new Date(startTime + durationInMs);
	ns.print(`[${dateToString(startTime)}] Expecting result at: ${dateToString(expectedResultAt)}.`);
}

export function dateToString(ms) {
	return new Date(ms).toLocaleString(["de-DE"]);
}



export function stringEquals(s1, s2) {
	return new String(s1).valueOf() == new String(s2).valueOf()
}
```

If I remember correctly, the problem first occured after updating to V. 1.4.0.

# include (where relevant)

- [X] Save file
[bitburnerSave_1642618805_BN4x0.json.txt](https://github.com/danielyxie/bitburner/files/7899919/bitburnerSave_1642618805_BN4x0.json.txt)

- [X] Minimal scripts to reproduce the issue
see description.

- [X] Steps to reproduce
Close the game. Load the game. --> Error occures.

- [X] Version of the game, e.g. Bitburner v1.3.0 (216bf616)
Bitburner v1.4.0 (1e126d40)

**Comment by smolgumball** [comment]
have you tried killing all scripts and restarting? seems the changes to how imports are resolved may have broken existing references to scripts in saved games.

**Comment by arnehaber** [comment]
Yes, I tried that and it temporary solves the problem. During the next start, the same problem occurs again.

**Comment by Dumpshock** [comment]
> I encountered the same issue. to resolve this, i had to change all imports to absolute (i.e. `/utils.js` instead of `utils.js`) for all scripts on `home`. all scripts on other servers don't work with `/utils.js` but with `utils.js`. very confusing.

I'm not seeing this behavior.  Currently all of my non-home scripts are generating this error whether I use '/library.js' or 'library.js' for the import.  Same behvior on home as well.

I haven't seen an absolute work unless the script I was calling was behind a directory.

Currently the only workaround I've had is to either keep the game running, or ensure that scripts are killed using the flag in the options during save.

**Comment by KonscienceGit** [comment]
Having a similar issue.
Can be reproduced at will on my save:
[bitburnerSave_1642713280_BN1x0.json.txt](https://github.com/danielyxie/bitburner/files/7908551/bitburnerSave_1642713280_BN1x0.json.txt)
How to reproduce:
1. from 'home', run deploy.js
2. restart the game without killing scripts.

Regarding c-goldschmidt's comment, any matter of paths expression (utils.js, /utils.js, ./utils.js) cause the issue on both 'home' and other servers.

**Comment by KonscienceGit** [comment]
It seems to be fixed (on my end) on v1.4.0 (49de4a28) that was deployed a couple minutes ago.
